### DISCUSSION TITLE
What about the thumbnails?

### DISCUSSION THREAD

**Discussion by ricassiocosta** [Arctosoft/Valv-Android#50] [Q&A] [answered]
Are the thumbnails stored somewhere (encrypted or decrypted)? Or are they generated in runtime and kept in memory? My feeling by using the app is the second option, but just asking to be sure.

**Comment by hej2010** [comment, marked as answer]
When you import a file it creates two files; one full size and one smaller scaled down thumbnail. Both are encrypted and stored in the folder you selected when importing.
When you view a file/folder it temporarily decrypts the files into the app's cache directory for faster loading. The cached files are deleted when you lock the vault or restart the app
